From c5b91ec63ba3e60d26d3864f92119869b89baf21 Mon Sep 17 00:00:00 2001
From: HassanMirza01 <hassanmirza01@hotmail.com>
Date: Tue, 20 Feb 2018 11:03:16 -0500
Subject: [PATCH] "0001-libutils-makeup-threadname-if-one-is-not-sent
 _______________________________

*Previous From cec5b4ca22e9bc76b0f4e07908e6c2a7f80b6171 Mon Sep 17 00:00:00 2001
*Previous From: Meticulus <theonejohnnyd@gmail.com>
*Previous Date: Sat, 3 Feb 2018 07:48:36 -0600
*Previous Subject: [PATCH] "0001-libutils-makeup-threadname-if-one-is-not-sent
*_______________________________
*
**Previous From 515668b788d04996fb48399228bd85ad793a9126 Mon Sep 17 00:00:00 2001
**Previous From: Meticulus <theonejohnnyd@gmail.com>
**Previous Date: Sun, 26 Mar 2017 11:28:07 -0500
**Previous Subject: [PATCH] libutils: makeup threadname if one is not sent
**
**Change-Id: Ia35caaa415a4e89a621a0ef867e05df463f6967f"
*
*Change-Id: If9b7caa1e9d739069dacc9dedd18a4a6a70766a2"

Change-Id: I3e3ba21b6a54927a22549e4517361b8e54f840bb
---
 libutils/Android.bp  |  1 -
 libutils/Threads.cpp | 20 ++++++++++----------
 2 files changed, 10 insertions(+), 11 deletions(-)

diff --git a/libutils/Android.bp b/libutils/Android.bp
index 91ca0ac..994b0a6 100644
--- a/libutils/Android.bp
+++ b/libutils/Android.bp
@@ -75,7 +75,6 @@ cc_library {
         "misc.cpp",
     ],
 
-    cflags: ["-Werror"],
     include_dirs: ["external/safe-iop/include"],
     header_libs: [
         "libutils_headers",
diff --git a/libutils/Threads.cpp b/libutils/Threads.cpp
index 2d7e46f..c5c2ec2 100644
--- a/libutils/Threads.cpp
+++ b/libutils/Threads.cpp
@@ -661,14 +661,14 @@ status_t Thread::readyToRun()
 
 status_t Thread::run(const char* name, int32_t priority, size_t stack)
 {
-    if (name == nullptr) {
-        ALOGW("Thread name not provided to Thread::run");
-        name = 0;
-    }
-
-    Mutex::Autolock _l(mLock);
-
-    if (mRunning) {
+    char nname[255];
+    if(name == nullptr)
+	sprintf(nname,"meticulus_%d",rand());
+    else
+	sprintf(nname,"%s",name);
+     Mutex::Autolock _l(mLock);
+ 
+     if (mRunning) {
         // thread already started
         return INVALID_OPERATION;
     }
@@ -687,10 +687,10 @@ status_t Thread::run(const char* name, int32_t priority, size_t stack)
     bool res;
     if (mCanCallJava) {
         res = createThreadEtc(_threadLoop,
-                this, name, priority, stack, &mThread);
+                this, nname, priority, stack, &mThread);
     } else {
         res = androidCreateRawThreadEtc(_threadLoop,
-                this, name, priority, stack, &mThread);
+                this, nname, priority, stack, &mThread);
     }
 
     if (res == false) {
-- 
2.14.1

